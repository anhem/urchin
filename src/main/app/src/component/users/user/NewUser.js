import React, {Component} from "react";
import history from '../../../history'

class NewUser extends Component {

    update = (e) => {
        let data = {
            field: e.target.name,
            value: e.target.value
        };

        this.props.callbacks.setUser(data);
    };

    create = () => {
        this.props.callbacks.createUser(this.props.user);
    };

    cancel = () => {
        history.push('/users');
    };

    render() {
        let username = this.props.user.username || '';
        let password = this.props.user.password || '';
        return (
            <div>
                <h2>User</h2>
                <input
                    name="username"
                    type="text"
                    placeholder="username"
                    value={username}
                    onChange={this.update}
                />
                <input
                    name="password"
                    type="password"
                    placeholder="password"
                    value={password}
                    onChange={this.update}
                />
                <button onClick={this.create}>Create User</button>
                <button onClick={this.cancel}>Cancel</button>
            </div>
        )
    }
}

export default NewUser