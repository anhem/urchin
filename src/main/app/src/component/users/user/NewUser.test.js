import React from 'react';
import {mount, shallow} from 'enzyme'
import toJson from 'enzyme-to-json';
import NewUser from "./NewUser";

describe('NewUser', () => {

    let props = {
        user: {},
    };

    it('renders without crashing', () => {
        expect(mount(<NewUser {...props}/>).length).toEqual(1);
    });

    it('match snapshot', () => {
        expect(toJson(shallow(<NewUser {...props}/>))).toMatchSnapshot();
    });
});
