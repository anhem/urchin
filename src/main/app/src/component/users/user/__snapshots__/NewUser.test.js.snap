// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewUser match snapshot 1`] = `
<div>
  <h2>
    User
  </h2>
  <input
    name="username"
    onChange={[Function]}
    placeholder="username"
    type="text"
    value=""
  />
  <FieldError
    field="username"
  />
  <input
    name="password"
    onChange={[Function]}
    placeholder="password"
    type="password"
    value=""
  />
  <FieldError
    field="password"
  />
  <button
    onClick={[Function]}
  >
    Create User
  </button>
  <button
    onClick={[Function]}
  >
    Cancel
  </button>
</div>
`;
